staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -575,21 +576,21 @@ func (ca *ConnectionAgent) syncPreExistingRemoteIfcs() error {
 							i)
 						time.Sleep(netFabricRetryPeriod)
 					}
-					// Do not run the PostDeleteExec, it probably
-					// would not work because the PostCreateExec ran
-					// in a different container if at all.
+					ca.unsetLocalAttState(nsn)
```

Comment:
Unrelated to immutability: this instruction applies to an attachment that transitions from local to remote during the initial interfaces sync from the connection agent. It unsets from an attachment's state its old local interface which has been deleted and replaced by a new remote interface during the initial sync. There was a comment from you asking why it was missing. You had gotten it right, its absence caused a bug. If that instruction is missing and the attachment is updated and a new interface is needed, the worker processing the attachment finds the old local interface in the attachment state and deletes it. The interface has already been deleted and the operation fails silently. The problem is that deletion of old/stale interfaces was mutually exclusive between local and remote interfaces: if a stale local interface was found it was assumed that there could not be also a stale remote interface for the same attachment. This mixed with the absence of the function this comment is about caused interfaces leaks.

Reply:
> I also noticed that all the commentary about the bool returned from updateVNStateForExistingAtt is wrong, it speaks of false where true is meant.

Dang, thanks for pointing it out. Fixed in 7690042.

Reply:
> Yes.
> Reviewing this reminded me of how complicated this logic is. I am re-motivated to look at simplifying it, in a later PR.

Are you referring to the whole CA or the initial sync up only? I agree that the whole CA is complicated, I think the sync up logic is not that difficult if properly explained (maybe better comments could be added) but is not apparent from that code. As I said while I was doing this and other PRs I stashed away unrelated tweaks to all controllers, I was thinking about merging them in a PR after this one. A considerable chunk of the CA tweaks is about the initial sync up (but they are not simplifications). That PR review might be the place to discuss changes. I might be wrong about this, but I fear there's not much complexity we can get rid of in that code.

Reply:
I think the whole CA is more complicated than it needs to be, including the initial sync.